cylc/flow/suite_srv_files_mgr.py

```diff
@@ -33,642 +34,692 @@
+            proc.terminate()
+        sleep(0.1)
+    fname = get_contact_file(reg)
+    ret_code = proc.wait()
+    out, err = (f.decode() for f in proc.communicate())
+    if ret_code:
+        LOG.debug("$ %s  # return %d\n%s", ' '.join(cmd), ret_code, err)
+    for line in reversed(out.splitlines()):
+        if line.strip() == old_proc_str:
+            # Suite definitely still running
+            break
+        elif line.split(None, 1)[0].strip() == "PID":
+            # Only "ps" header - "ps" has run, but no matching results.
+            # Suite not running. Attempt to remove suite contact file.
+            try:
+                os.unlink(fname)
+                return
+            except OSError:
+                break
+
+    raise SuiteServiceFileError(
+        CONTACT_FILE_EXISTS_MSG % {
+            "host": old_host,
+            "port": old_port,
+            "pid": old_pid_str,
+            "fname": fname,
+            "suite": reg,
+        }
+    )
+
+
+def dump_contact_file(reg, data):
+    """Create contact file. Data should be a key=value dict."""
+    # Note:
+    # 1st fsync for writing the content of the contact file to disk.
+    # 2nd fsync for writing the file metadata of the contact file to disk.
+    # The double fsync logic ensures that if the contact file is written to
+    # a shared file system e.g. via NFS, it will be immediately visible
+    # from by a process on other hosts after the current process returns.
+    with open(get_contact_file(reg), "wb") as handle:
+        for key, value in sorted(data.items()):
+            handle.write(("%s=%s\n" % (key, value)).encode())
+        os.fsync(handle.fileno())
+    dir_fileno = os.open(get_suite_srv_dir(reg), os.O_DIRECTORY)
+    os.fsync(dir_fileno)
+    os.close(dir_fileno)
+
+
+def get_contact_file(reg):
+    """Return name of contact file."""
+    return os.path.join(
+        get_suite_srv_dir(reg), SuiteFiles.Service.CONTACT)
+
+
+def get_auth_item(item, reg, owner=None, host=None, content=False):
+    """Locate/load passphrase, SSL private key, SSL certificate, etc.
+
+    Return file name, or content of file if content=True is set.
+    Files are searched from these locations in order:
+
+    1/ For running task jobs, service directory under:
+       a/ $CYLC_SUITE_RUN_DIR for remote jobs.
+       b/ $CYLC_SUITE_RUN_DIR_ON_SUITE_HOST for local jobs or remote jobs
+          with SSH messaging.
+
+    2/ For suite on local user@host. The suite service directory.
+
+    3/ Location under $HOME/.cylc/ for remote suite control from accounts
+       that do not actually need the suite definition directory to be
+       installed:
+       $HOME/.cylc/auth/SUITE_OWNER@SUITE_HOST/SUITE_NAME/
+
+    4/ For remote suites, try locating the file from the suite service
+       directory on remote owner@host via SSH. If content=False, the value
+       of the located file will be dumped under:
+       $HOME/.cylc/auth/SUITE_OWNER@SUITE_HOST/SUITE_NAME/
```

Comment:
Usage 3 and 4 should disappear under Cylc 8. Both use cases should be handled by the Hub-UIS-WFS interaction.

Reply:
Are these used by the GUI only?

Reply:
In Cylc 8, I believe we are making the assumption that only the main users on the shared file system will connect a client to the WFS directly. You will go through the Hub+UIS if:
* You are not the user running the WFS.
* You are not on a host that has a shared file system with the host running the WFS.

Reply:
Adding this as a TODO comment ...